documents/invoice/model.go

```diff
@@ -325,10 +330,11 @@ func (i *Invoice) UnpackCoreDocument(coreDoc *coredocumentpb.CoreDocument) error
 		i.InvoiceSalts = documents.ConvertToProofSalts(coreDoc.EmbeddedDataSalts)
 	}
 
-	i.CoreDocument = new(coredocumentpb.CoreDocument)
-	proto.Merge(i.CoreDocument, coreDoc)
-	i.CoreDocument.EmbeddedDataSalts = nil
-	i.CoreDocument.EmbeddedData = nil
+	i.CoreDocumentModel = new(documents.CoreDocumentModel)
```

Comment:
Same way we did pack, we can add `coreDocumentModel.UnpackCoreDocument(coreDocModel *documents.CoreDocumentModel)`.